conans/test/integration/toolchains/meson/test_test.py

```diff
@@ -0,0 +1,61 @@
+import os
+import pytest
+import textwrap
+
+from conans.test.assets.sources import gen_function_cpp
+
+
+from ._base import TestMesonBase
+
+
+@pytest.mark.tool_pkg_config
+class MesonTest(TestMesonBase):
+    _test_package_meson_build = textwrap.dedent("""
+    project('test_package', 'cpp')
+    hello = dependency('hello', version : '>=0.1')
+    test_package = executable('test_package', 'test_package.cpp', dependencies: hello)
+    test('test package', test_package)
+    """)
+
+    _test_package_conanfile_py = textwrap.dedent("""
+    import os
+    from conans import ConanFile
+    from conan.tools.meson import Meson, MesonToolchain
+
+
+    class TestConan(ConanFile):
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "pkg_config"
+        _meson = None
+
+        def generate(self):
+            tc = MesonToolchain(self)
+            tc.generate()
+
+        def _configure_meson(self):
+            if not self._meson:
+                self._meson = Meson(self)
+                self._meson.configure()
+            return self._meson
+
+        def build(self):
+            meson = self._configure_meson()
+            meson.build()
+
+        def test(self):
+            meson = self._configure_meson()
```

Comment:
If done ok it should be possible to do just ``Meson(self)`` here. No need to have a ``conafigure_meson()`` at all, that was a necessary thing for the old ``CMake`` build helper because it had a bug.

Reply:
I think we were using this pattern to avoid `meson configure` (or `cmake configure`) running twice, so we were caching an object to avoid extra configure invocation. I am fine to remove it for tests, as they are very short and simple, and don't take much time to configure. in real life, we have seen projects taking minutes to configure.

Reply:
removed `_configure_meson`